setup.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+#written by @initinfosec - 05 JUN 2020
+#simple bash script to install/configure necessary setup for noraj PWK/OSCP report md to PDF conversion tool
+
+#only tested on kali 2020.x - error checking/input validation is not thorough
+
+# check if running with sudo
+if [[ $EUID -ne 0 ]]; then
+	echo -e "Script will need sudo/root privs for portions of the install actions. You may be prompted for your password if installs need to be made.\n\n"
+	SUDO='sudo'
+fi
+
+# caveat emptor preface
+
+echo -e "This setup script only needs to be run if you have not installed LaTeX, Pandoc, etc. and configured according to https://github.com/initinfosec/OSCP-Exam-Report-Template-Markdown\nThis script was only tested on kali 2020.x\n\n"
+
+read -n1 -rsp "If already configured, press Q to exit script. Otherwise, press any key to continue: " key
+
+if [ "${key,,}" = "q" ]; then
+	echo -e "\nExiting.\n" && exit
+else
+	echo -e "\nContinuing with setup.\n"
+fi
+
+
+# search for 7zip, pandoc, LaTeX full, and eisvogel on system
+
+cwd="$(pwd)"
+eisvogelDir="$(pwd)/eisvogel"
+
+if which 7z &> /dev/null ; then
+	echo -e "7z detected installed, moving on.\n"
+else
+	echo -e "7z not detected, installing.\n"
+	yes | $SUDO apt install p7zip-full &> /dev/null && echo -e "p7zip-full installed.\n"
```

Comment:
I'm not a fan of making a requirement script because it's always distro specific.

Reply:
that's fair, i was operating under the assumption almost all users would be building the report from kali, but you're correct, thanks for the review.

Reply:
FYI I did the OSCP and PWK with BlackArch Linux, also Offensive Security just **recommend** to use Kali (because it's their OS) to be able to provide support but anyone is free to use any OS during the exam.